flit_core/flit_core/common.py

```diff
@@ -66,7 +66,7 @@ def __init__(self, name, directory=Path()):
     @property
     def file(self):
         if self.is_package:
-            return self.path / '__init__.py'
+            return self.path.joinpath('__init__.py')
```

Comment:
Could you please remove these `/` -> joinpath changes? I don't think they're particularly beneficial, and seem to be the vast majority of this PR.

Reply:
I changed that part to ensure `Path` types were always being used in those locations properly, the implementation between `/` and joinpath appears to be [slightly different](https://github.com/python/cpython/blob/v3.10.2/Lib/pathlib.py#L843-L861). I find joinpath to generally be preferable as it's more explicit but that's somewhat subjective.

Reply:
Well, it adds a lot of noise to this PR. I'm unlikely to re-review this PR, if this style of change isn't dropped. :)

Reply:
Ok, I'll back those parts out.

Reply:
> Could you please remove these `/` -> joinpath changes?

I think I have all of these that weren't needed backed out now.